delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/migration/MigrationV16ToV17.scala

```diff
@@ -0,0 +1,178 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.migration
+
+import akka.actor.typed.ActorSystem
+import akka.stream.alpakka.cassandra.scaladsl.CassandraSession
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.ElasticSearchViews
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.migration.MigrationV16ToV17._
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewEvent.{ElasticSearchViewCreated, ElasticSearchViewUpdated}
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewValue.IndexingElasticSearchViewValue
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.{ElasticSearchViewEvent, ElasticSearchViewType, ElasticSearchViewValue}
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.JsonLdContext.keywords
+import ch.epfl.bluebrain.nexus.delta.sdk.migration.Migration
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sdk.views.pipe._
+import ch.epfl.bluebrain.nexus.delta.sourcing.config.CassandraConfig
+import ch.epfl.bluebrain.nexus.delta.sourcing.utils.CassandraUtils
+import com.datastax.oss.driver.api.core.cql.PreparedStatement
+import com.typesafe.scalalogging.Logger
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.semiauto.deriveConfiguredEncoder
+import io.circe.parser._
+import io.circe.syntax.EncoderOps
+import io.circe.{Encoder, Error, Printer}
+import monix.bio.Task
+
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+import java.util.UUID
+import scala.annotation.nowarn
+
+/**
+  * Migrate elasticsearch views to new model with pipeline
+  */
+final class MigrationV16ToV17 private (
+    session: CassandraSession,
+    config: CassandraConfig
+) extends Migration {
+
+  private val selectViewEvents =
+    s"SELECT persistence_id, event, sequence_nr, timestamp FROM ${config.keyspace}.tag_views where tag_name = ? limit 1000000 allow filtering;"
+
+  private val updateMessage: String =
+    s"UPDATE ${config.keyspace}.messages set event = ? where persistence_id = ? and partition_nr = 0 and sequence_nr = ? and timestamp = ?"
```

Comment:
is the partition number always 0?

Reply:
For us, yes.

```
The messages table that stores the events is partitioned by (persistence_id, partition_nr). The partition_nr is an artificial partition key to ensure that the Cassandra partition does not get too large if there are a lot of events for a single persistence_id.

akka.persistence.cassandra.journal.target-partition-size controls the number of events that the journal tries to put in each Cassandra partition. It is a target as persistAll calls will have all the events in the same partition even if it will exceed the target partition size to ensure atomicity.

It is not possible to change the value once you have data so consider if the default of 500000...
```

Reply:
is it possible to read it from the table and write it back with the same value, or exclude it from the update statement?

Reply:
We can't as it is part of the PK